Test/fixture/norm/sop/Expected1.java

```diff
@@ -0,0 +1,18 @@
+public class C
+{
+  public static void main(String[] args)
+  {
+    {
+      int i, j, k;
+      i = j + k + i;
+    }
+    {
+      int i, j, k;
+      i = j*k + j*j + k*k + k*j;
+    }
+    {
+      int i, j, k;
+      i = j*j*k*k*k + j*j*1*k*k + j*k;
```

Comment:
This is nice.

A next step (other normalization) is to get rid of identities `* 1` and `+ 0`.

Also:
+ move constants to the left hand side of `+, *`,
+ normalize `x << 1 => x * 2`,
+ constant folding.

**Note:** this is not a request for changes =)